### PR TITLE
Use topics enum instead of tuples

### PR DESCRIPTION
**Pull request by m4ra**
**Tasks**
- [ ] PR name contains story or task reference
- [ ] Documentation (docs and inline)
- [ ] Tests (including n+1 and django_assert_num_queries where applicable)
- [ ] Changelog

### CODE REVIEW

File: adhocracy4/projects/migrations/0043_migrate_topics_to_m2m_topics.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 
 
 def add_topics_to_m2m_table(apps, schema_editor):
     if hasattr(settings, "A4_PROJECT_TOPICS"):
-        topicsenum = settings.A4_PROJECT_TOPICS
+        # get the module from the settings string path without the class
+        index = settings.A4_PROJECT_TOPICS.rfind(".")
+        # get the class name from the settings string path
+        module = importlib.import_module(settings.A4_PROJECT_TOPICS[:index])
+        topicsenum = getattr(module, settings.A4_PROJECT_TOPICS[index + 1 :])
```
Comment (goapunk) [inline adhocracy4/projects/migrations/0043_migrate_topics_to_m2m_topics.py:15]: This seems a bit complicated and raises the question whether we need the `name` of the topic in the database. For our purposes the `code` would be enough, right?

And if we need the label (e.g. in the template or via the api) we can get it via `topicsenum(code).label`, for example in the serializer or by overwriting the `def _str_` of the Topic model.

Comment (m4ra) [reply adhocracy4/projects/migrations/0043_migrate_topics_to_m2m_topics.py:15]: I agree, we don't need the name. I will remove it then.

File: adhocracy4/projects/models.py [hunk 2]
```diff
@@ -25,13 +26,12 @@
...
-    name = models.CharField(
-        max_length=120,
-        verbose_name=_("Topic"),
-    )
 
     def __str__(self):
-        return self.name
+        if hasattr(settings, "A4_PROJECT_TOPICS"):
+            topics_enum = import_string(settings.A4_PROJECT_TOPICS)
+            return str(topics_enum(self.code).label)
```
Comment (m4ra) [inline adhocracy4/projects/models.py:33]: Nice!

**Review by goapunk — APPROVED** [review]
LGTM!

**Pull request merged**